wgengine/router_freebsd.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package wgengine
+
+import (
+	"github.com/tailscale/wireguard-go/device"
+	"github.com/tailscale/wireguard-go/tun"
+	"tailscale.com/logger"
+)
+
+type freebsdRouter struct {
+	tunname string
+}
+
+func NewUserspaceRouter(logf logger.Logf, tunname string, dev *device.Device, tuntap tun.Device, netChanged func()) Router {
+	r := freebsdRouter{
+		tunname: tunname,
+	}
+	return &r
+}
+
+func (r *freebsdRouter) Up() error {
+	return nil
+}
+
+func (r *freebsdRouter) SetRoutes(rs RouteSettings) error {
+	if SetRoutesFunc != nil {
+		return SetRoutesFunc(rs)
```

Comment:
Don't use SetRoutesFunc. It's used only by the mac/ios builds, as a way to bridge Go and Swift together. `router_freebsd.go` should implement Up() and SetRoutes() in terms of FreeBSD APIs or commands (e.g. `ifconfig`, `route`, or equivalent kernel APIs, whatever's idiomatic for the platform).

Reply:
(to be clear, this is a bug on our end - SetRoutesFunc exists in the cross-platform code as a hack that enables us to build bits of the mac/ios code from Linux, to reduce the amount of breakage - but it's not clearly documented that this is what's happening)